### PR TITLE
feat(featureflag): adding in feature flag baseline

### PR DESCRIPTION
**Pull request by bassrock**
## Summary

Add in the ability to use feature flags in the app.

Usage:

```swift

let assigned = Services.shared.featureFlagService.isAssigned(flag: "temp-ios.test", variant: "variant b")

//then track the user felt it
Services.shared.featureFlagService.trackFeatureFlagFelt(flag: "temp-ios.test", variant: "variant b")

```

## Test Steps
* See feature flags are loaded via the console.
* A debug menu for iOS to come with this capability :) 

## PR Checklist:
- [ ] Added Unit / UI tests
- [ ] Self Review (review, clean up, documentation, run tests)
- [ ] Basic Self QA

### CODE REVIEW

File: PocketKit/Sources/Sync/PocketSource.swift [hunk 1]
```diff
@@ -72,13 +73,13 @@ public class PocketSource: Source {
...
         user: User,
-        sessionProvider: SessionProvider,
+        appSession: AppSession,
         consumerKey: String,
         defaults: UserDefaults,
         backgroundTaskManager: BackgroundTaskManager
     ) {
         let apollo = ApolloClient.createDefault(
-            sessionProvider: sessionProvider,
+            sessionProvider: appSession as! SessionProvider,
```
Comment (bassrock) [inline PocketKit/Sources/Sync/PocketSource.swift:82]: This does not need to be force cast in Services, but for some reason it does here. And it works. @dskuza  may have insights.

File: PocketKit/Sources/Sync/Space/Space.swift [hunk 2]
```diff
@@ -37,9 +37,7 @@ public class Space {
     func delete(_ object: NSManagedObject, in context: NSManagedObjectContext? = nil) {
         let context = context ?? backgroundContext
         context.performAndWait {
-            guard let object = backgroundObject(with: object.objectID) else {
-                return
-            }
+            let object = context.object(with: object.objectID)
```
Comment (bassrock) [inline PocketKit/Sources/Sync/Space/Space.swift:40]: @Gio2018 opened a pr with this separately https://github.com/Pocket/pocket-ios/pull/575

File: PocketKit/Sources/PocketKit/Refresh/FeatureFlagsRefreshCoordinator.swift [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import Sync
+import SharedPocketKit
+import Combine
+
+/// Refresh coordinator to handle the refreshing of feature flags
+class FeatureFlagsRefreshCoordinator: RefreshCoordinator {
+
+    let taskID: String = "com.mozilla.pocket.refresh.featureFlags"
+
+    let refreshInterval: TimeInterval? = 60 * 60 * 48 // once every 48 hours.
```
Comment (Gio2018) [inline PocketKit/Sources/PocketKit/Refresh/FeatureFlagsRefreshCoordinator.swift:15]: why setting this as an optional here and then force unwrap it below? Can it just be a constant?

File: PocketKit/Sources/PocketKit/Refresh/UserRefreshCoordinator.swift [hunk 4]
```diff
@@ -34,6 +34,7 @@ class UserRefreshCoordinator: RefreshCoordinator {
     func refresh(isForced: Bool = false, _ completion: @escaping () -> Void) {
         Task {
```
Comment (Gio2018) [inline PocketKit/Sources/PocketKit/Refresh/UserRefreshCoordinator.swift:35]: capturing self explicitly might not be strictly required if we are sure that the task completes (although a weak capture is a failsafe that always ensures that self is not retained more than it should), but curious if there's a specific reason why it's not captured here as opposed to other places where it's captured?

File: PocketKit/Sources/Sync/PocketSource.swift (hunk 1 above)
Comment (Gio2018) [reply PocketKit/Sources/Sync/PocketSource.swift:82]: it feels like we have protocols, types and extensions defined in different modules that also sometimes depend on each other, that might be the cause of this?

**Review by Gio2018 — APPROVED** [review]
Looks good @bassrock ! Just a few non blocking questions!

File: PocketKit/Sources/PocketKit/FeatureFlagService.swift [hunk 5]
```diff
@@ -0,0 +1,38 @@
+//
+//  File.swift
+//  
+//
+//  Created by Daniel Brooks on 11/22/22.
+//
```
[the remark this answers is not in the document]
Comment (bassrock) [reply PocketKit/Sources/PocketKit/FeatureFlagService.swift:6]: ugh. i wish xcode auto added this.

File: PocketKit/Sources/PocketKit/Refresh/UserRefreshCoordinator.swift (hunk 4 above)
Comment (bassrock) [reply PocketKit/Sources/PocketKit/Refresh/UserRefreshCoordinator.swift:35]: i dumb.

**Pull request merged**